src/sync.js

```diff
@@ -496,11 +505,46 @@ SyncApi.prototype.retryImmediately = function() {
     return true;
 };
 
+SyncApi.prototype._syncFromCache = async function(savedSync) {
```

Comment:
jsdoc wouldn't hurt, particularly since you've [already written it](https://github.com/matrix-org/matrix-js-sdk/pull/601/files#diff-b5ce4044fcc6267228c6281fd7483b4cR545)